DTS-STN/Alpha-Site
# cypress-axe and jest-axe

# Description

[Setup automated a11y testing](https://trello.com/c/JO8M1zcU/63-set-up-automated-accessibility-testing)

This pr includes [cypress-axe](https://github.com/component-driven/cypress-axe) and [jest-axe](https://github.com/nickcolley/jest-axe), configuration to support the reporting of a11y failures when running cypress tests and github workflow actions to support the testing in CI.

Additionally, a [custom NextJS Document](https://nextjs.org/docs/advanced-features/custom-document) has been created to allow for control of the html lang attribute and other html or body element attributes that the application needs.

## Acceptance Criteria

Unit and a11y tests can be automated in the CI.

## Test Instructions

1. `docker-compose up dev`
2. `npm run cypress:run`
3. See that tests pass
4. Remove the `lang` attribute on the `<Html />` element found in /pages/_document.js and save
5. `npm run cypress:run`
5. See that test fails

## Checklist


## Product and Sprint Backlog

[Trello](https://trello.com/b/ZqWtJSyh/alpha-site-board)
